fiat-roles/src/main/java/com/netflix/spinnaker/fiat/permissions/RedisPermissionsRepository.java

```diff
@@ -206,40 +288,36 @@ public RedisPermissionsRepository put(@NonNull UserPermission permission) {
 
   private Optional<UserPermission> getFromRedis(@NonNull String id) {
     try {
+      TimeoutContext timeoutContext =
+          new TimeoutContext(
+              String.format("getPermission for user: %s", id),
+              clock,
+              configProps.getRepository().getGetPermissionTimeout());
       boolean userExists =
-          UNRESTRICTED.equals(id)
-              || redisClientDelegate.withCommandsClient(
-                  c -> {
-                    return c.sismember(allUsersKey(), id);
-                  });
+          UNRESTRICTED.equals(id) || redisRead(timeoutContext, c -> c.sismember(allUsersKey(), id));
       if (!userExists) {
         return Optional.empty();
       }
-      final RawUserPermission userResponseMap = new RawUserPermission();
-      redisClientDelegate.withMultiKeyPipeline(
-          p -> {
-            resources.stream()
-                .map(Resource::getResourceType)
-                .forEach(
-                    r -> {
-                      String userKey = userKey(id, r);
-                      Response<Map<String, String>> resourceMap = p.hgetAll(userKey);
-                      userResponseMap.put(r, resourceMap);
-                    });
-            Response<Boolean> admin = p.sismember(adminKey(), id);
-            p.sync();
-            userResponseMap.isAdmin = admin.get();
+      UserPermission userPermission = new UserPermission().setId(id);
+      resources.forEach(
```

Comment:
I don't think the pipeline to fetch all the things and then deserialize later would be meaningfully faster than just executing the appropriate calls, and this has the added benefit of only retrying some portion of the call that failed due to a jedis exception instead of re-invoking the whole pipeline